gateway/middleware.go

```diff
@@ -407,6 +407,14 @@ func (t BaseMiddleware) ApplyPolicies(session *user.SessionState) error {
 							}
 						}
 
+						for _, t := range v.AllowedTypes {
+							for ri, rt := range r.AllowedTypes {
+								if t.Name == rt.Name {
```

Comment:
Can the name match multiple times? So we cant break after the first match?
And similarly, would it be worth using a map or something where the keys are the name so we don't have to nest loops?

Reply:
I think redundant names should be filtered somewhere else. For example, we can run a sanitizer before saving the configuration to filter the redundant names, if this is a problem. 

> And similarly, would it be worth using a map or something where the keys are the name so we don't have to nest loops?

Two levels of nested loops are just acceptable, I think.